### PR TITLE
Add term-project.el, which is implementation for project.el

### PR DESCRIPTION
**Pull request by ROCKTAKEY**
Hi, thanks for the great package!

I would like to use this package with project.el, so I add term-project.el.
Thanks to your abstraction, almost all functions are changed little compared to term-projectile.el.

I also added some docstrings and make some small change to pass checkdoc and package-lint so that term-project.el can be added to MELPA smoothly.

Or, if you don't want to maintenance term-project.el, could you allow me to manage term-project.el package in my repository? Since there is no LICENSE file and term-project.el is made largely from term-projectile.el, I cannot do that without your permission.

### CODE REVIEW

**Comment by colonelpanic8** [comment]
> Or, if you don't want to maintenance term-project.el, could you allow me to manage term-project.el package in my repository? Since there is no LICENSE file and term-project.el is made largely from term-projectile.el, I cannot do that without your permission.

Happy to go whichever route makes the most sense. If you want my opinion, i think having it in this repository probably makes it easier to avoid breaking compatibilty and stuff.

File: term-project.el [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+  (term-project-global-switch :delta -1))
+
+;;;###autoload
+(defun term-project-global-create-new ()
+  "Make a new `ansi-term' buffer in `term-project-global-directory'."
+  (interactive)
+  (term-project-create-new term-project-global-directory))
+
+(provide 'term-project)
+;;; term-project.el ends here
```
Comment (colonelpanic8) [inline term-project.el:155]: a lot of this is really boilerplate. Perhaps we should write a macro that can generate all of these functions.

**Comment by ROCKTAKEY** [comment]
Thank a lot!
May I add this term-project to MELPA?

**Comment by colonelpanic8** [comment]
> Thank a lot!
> May I add this term-project to MELPA?

yes, of course.

**Pull request merged**